src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -195,7 +200,62 @@ internal string FullPath
             {
                 // Should already be canonicalized to a full path
                 ErrorUtilities.VerifyThrowInternalRooted(value);
-                _solutionFile = value;
+                if (FileUtilities.IsSolutionFilterFilename(value))
+                {
+                    try
+                    {
+                        using JsonDocument text = JsonDocument.Parse(File.ReadAllText(value));
+                        JsonElement solution = text.RootElement.GetProperty("solution");
+                        _solutionFile = solution.GetProperty("path").GetString();
+                        if (!File.Exists(_solutionFile))
+                        {
+                            ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile
+                            (
+                                false /* just throw the exception */,
+                                "SubCategoryForSolutionParsingErrors",
+                                new BuildEventFileInfo(_solutionFile),
+                                "SolutionFilterMissingSolutionError",
+                                value,
+                                _solutionFile
+                            );
+                        }
+                        _solutionFilter = new HashSet<string>(NativeMethodsShared.IsLinux ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
+                        foreach (JsonElement project in solution.GetProperty("projects").EnumerateArray())
+                        {
+                            if (!File.Exists(Path.Combine(Path.GetDirectoryName(_solutionFile), project.GetString())) && !File.Exists(project.GetString()))
```

Comment:
Why not defer the check to the normal time that the solution would deal with the file?

I would expect an error when the filter's project path isn't a project in the `.sln` file (that's `SolutionFilterFilterContainsProjectNotInSolution`, right?), but not when the project doesn't exist on disk (at this point).

Reply:
Because this is a problem with the .slnf, not with the .sln. You're right that that other error is for if there's something in the .slnf that isn't in the .sln.

I think it's valuable to have both types of exceptions. If we let projects exist in the .slnf that don't exist elsewhere, the user could try to modify the .slnf, make a small typo, and be very confused as to why their project wasn't being built.

Reply:
I don't understand what you're saying here. Can you clarify?

The slnf filters the solution, right? So the entries in the slnf should match the projects in the solution. I don't see what presence/absence on disk has to do with anything.

Reply:
The motivating scenario here is if the user wants to include a project in what is being built, so they add it to the .slnf but make a typo. This throws an error if they do that.

Another way of looking at it is that if a file doesn't exist, we can't build it anyway, so it shouldn't be in the .slnf.